src/SFML/Graphics/Image.cpp

```diff
@@ -171,46 +172,43 @@ void Image::createMaskFromColor(const Color& color, Uint8 alpha)
 
 
 ////////////////////////////////////////////////////////////
-[[nodiscard]] bool Image::copy(const Image& source, const Vector2u& dest, const IntRect& sourceRect, bool applyAlpha)
+[[nodiscard]] bool Image::copy(const Image& source, const Vector2u& destination, const IntRect& sourceRect, bool applyAlpha)
 {
     // Make sure that both images are valid
     if (source.m_size.x == 0 || source.m_size.y == 0 || m_size.x == 0 || m_size.y == 0)
         return false;
 
-    // Make sure the sourceRect components are non-negative before casting them to unsigned values
-    if (sourceRect.left < 0 || sourceRect.top < 0 || sourceRect.width < 0 || sourceRect.height < 0)
+    IntRect absRect(sourceRect.getAbsoluteRect());
+
+    // Make sure the absolute position of sourceRect is not negative
+    if (absRect.position.x < 0 || absRect.position.y < 0)
         return false;
 
-    Rect<unsigned int> srcRect(sourceRect);
+    // If the provided source rectangle is empty, use the entire source image
+    if (absRect.size.x == 0 || absRect.size.y == 0)
+        absRect = IntRect({0, 0}, Vector2i(source.m_size));
 
-    // Use the whole source image as srcRect if the provided source rectangle is empty
-    if (srcRect.width == 0 || srcRect.height == 0)
-    {
-        srcRect = Rect<unsigned int>({0, 0}, source.m_size);
-    }
     // Otherwise make sure the provided source rectangle fits into the source image
-    else
-    {
-        // Checking the bottom right corner is enough because
-        // left and top are non-negative and width and height are positive.
-        if (source.m_size.x < srcRect.left + srcRect.width || source.m_size.y < srcRect.top + srcRect.height)
-            return false;
-    }
+    else if (source.m_size.x < static_cast<unsigned int>(absRect.getRight()) ||
+             source.m_size.y < static_cast<unsigned int>(absRect.getBottom()))
+        return false;
 
     // Make sure the destination position is within this image bounds
-    if (m_size.x <= dest.x || m_size.y <= dest.y)
+    if (destination.x > m_size.x || destination.y > m_size.y)
         return false;
 
     // Then find the valid size of the destination rectangle
-    const Vector2u dstSize(std::min(m_size.x - dest.x, srcRect.width), std::min(m_size.y - dest.y, srcRect.height));
+    Vector2u dstSize(std::min(m_size.x - destination.x, static_cast<unsigned int>(absRect.size.x)),
+                     std::min(m_size.y - destination.y, static_cast<unsigned int>(absRect.size.y)));
 
     // Precompute as much as possible
-    const std::size_t  pitch     = static_cast<std::size_t>(dstSize.x) * 4;
-    const unsigned int srcStride = source.m_size.x * 4;
-    const unsigned int dstStride = m_size.x * 4;
+    std::size_t pitch     = dstSize.x * 4;
+    std::size_t srcStride = source.m_size.x * 4;
```

Comment:
I agree with using `std::size_t` here, but I think the `const` was valid here

Reply:
Agreed. Done